canonical/maas-ui
# refactor(machines): move machine select table to a reusable component

## Done

- Make the machine selection table a reusable component.
- Migrate tests to RTL.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Open [CloneFormFields.tsx](https://github.com/canonical/maas-ui/blob/3ea99f1d1202cd425eb5e59b2ad4d84e2c34b318/src/app/machines/components/MachineHeaderForms/ActionFormWrapper/CloneForm/CloneFormFields/CloneFormFields.tsx#L41) and change `const loadingMachines ...` to `const loadingMachines = true` (this isn't working as this form hasn't been updated to use the new API).
- Open [MachineSelectTable.tsx](https://github.com/canonical/maas-ui/blob/3ea99f1d1202cd425eb5e59b2ad4d84e2c34b318/src/app/base/components/MachineSelectTable/MachineSelectTable.tsx#L84) and change `const loadingMachines ...` to `const loadingMachines = true`.
- Start the UI.
- Go to the machine list and tick some machines to clone.
- In the take action menu select "Clone from..."
- In the form you should see some machines to clone from.
- Search and the list should get updated.
- Select a machine and the details should appear.

Fixes: canonical/app-tribe#1224.
